### ISSUE TITLE
ENH: Inconsistency when testings spin and seniority in ehCI wavefunctions

### ISSUE THREAD

**Issue by carlosevmoura** [mqcomplab/Fanpy#15]
### Describe the new feature:

The actual fanpy.tools.satisfies_conditions function cannot correctly assign valid spins and seniorities for SDs in extended-hierarchy CI wavefunctions.****

### How would you implement this feature?

I plan to add a flag as an argument in fanpy.tools.satisfies_conditions and fanpy.tools.sd_lists to hierarchy constructions and check conditions properly. It also requires changes in the ehCI wavefunctions actually implemented.

**Comment by rugwed-lokhande** [comment]
are you saying a flag like an "if" condition in satisfies_condition() to check if the wfn is ehci then do this?

**Comment by carlosevmoura** [comment]
Yes, exactly this. It is a solution for this problem. However, it can still be improved if the solution comes directly from the SD creations in the ehCI wavefunction.
